doc/website/getting-started/overview.md

```diff
@@ -0,0 +1,540 @@
+an option.
+
+We now describe the how to use the API in iceoryx applications. We will omit namespaces in several places to keep
+the code concise. In most cases it can be assumed that we are using namespace ``iox::cxx``. We also will use ``auto``
+sparingly to clearly show which types are involved, but in many cases automatic type deduction is possible and can
+shorten the code.
+
+### Optional
+
+The type ``cxx::optional<T>`` is used to indicate that there may or may not be a value of a specific type ``T``
+available. This is essentially the maybe monad in functional programming. Assuming we have some optional (usually the
+result of some computation)
+```
+optional<int> result = someComputation();
+```
+we can check for its value using
+```
+if(result.has_value())
+{
+    auto value = result.value();
+    //do something with the value
+} else {
+    //handle the case that there is no value
+}
+```
+A shorthand to get the value is 
+```
+auto value = *result;
+```
+
+Note that getting the value if there is no value is undefined behavior, so it must be checked beforehand.
+
+We can achieve the same with the functional approach by providing a function for both cases.
+
+```
+result.and_then([](int& value) { /*do something with the value*/ })
+      .or_else([]() { /*handle the case that there is no value*/ });
+```
+Notice that we get the value by reference, so if a copy is desired it has to be created explicitly in the lambda or
+function we pass.
+
+The optional can be be initialized from a value directly
+```
+optional<int> result = 73;
+result = 37;
+```
+If it is default initialized it is automatically set to its null value of type ``iox::cxx::nullopt_t``;
+This can be also done directly by using the constant ``iox::cxx::nullopt``
+
+```
+result = iox::cxx::nullopt;
+```
+
+### Expected
+``iox::cxx::expected<T, E>`` generalizes ``iox::cxx::optional`` by admitting a value of another type ``E`` instead of
+no value at all, i.e. it contains either a value of type ``T`` or ``E`` (and roughly corresponds to the either
+monad). This is usually used to pass a value of type ``T`` or an error that may have occurred, i.e. ``E`` is the
+error type. For more information on how it is used for error handling see
+[error-handling.md](../../design/error-handling.md).
+
+Assume we have ``E`` as an error type, then we can create a value
+```
+iox::cxx::expected<int, E> result(iox::cxx::success<int>(73));
+```
+
+and use the value or handle a potential error
+```
+if (!result.has_error())
+{
+    auto value = result.value();
+    // do something with the value
+}
+else
+{
+    auto error = result.get_error();
+    // handle the error
+}
+```
+
+Should we need an error value we set
+```
+result = iox::cxx::error<E>(errorCode);
+```
+which assumes that E can be constructed from an ``errorCode``.
+
+We again can employ a functional approach like this
+```
+auto handleValue = [](int& value) { /*do something with the value*/ };
```

Comment:
Could you apply our clang-format?

Reply:
Again, it uglifies and lengthens the code...